inference-engine/src/gna_plugin/gna_graph_compiler.cpp

```diff
@@ -527,6 +527,64 @@ void GNAGraphCompiler::finalizeConvolution1DPrimitive(InferenceEngine::CNNLayerP
 
 #if GNA_LIB_VER == 2
 
+void CheckCnn2DInputSupported(const unsigned h, const unsigned w) {
+    if (h < 16 || h > 384) {
+        THROW_GNA_EXCEPTION << "Unsupported input height: " << h << ", must be in range <16, 384>";
+    }
+    if (w < 16 || w > 240) {
+        THROW_GNA_EXCEPTION << "Unsupported input width: " << w << ", must be in range <16, 240>";
+    }
+}
+
+struct Cnn2DKernelSizeLimit {
+    unsigned squareLimit;
+    unsigned hVecLimit;
+    unsigned vVecLimit;
+};
+
+Cnn2DKernelSizeLimit GetCnn2DLimit(OvGnaType inPrecision, unsigned channels) {
+    Cnn2DKernelSizeLimit l337{ 3, 3, 7 };
+    Cnn2DKernelSizeLimit l227{ 2, 2, 7 };
+    Cnn2DKernelSizeLimit vertVec7{ 1, 1, 7 };
+
+    struct Cnn2DLimit {
+        unsigned channelLimit;
+        Cnn2DKernelSizeLimit smallChannel;
+        Cnn2DKernelSizeLimit bigChannel;
+    };
+
+    Cnn2DLimit l16{ 120, l337, vertVec7 };
+    Cnn2DLimit l8{ 240, l337, l227 };
+    const auto& l = inPrecision == OvGnaTypeInt8 ? l8 : l16;
+    return (channels <= l.channelLimit ? l.smallChannel : l.bigChannel);
+}
+
+void ExpectCnn2DKernelValid(unsigned kW, unsigned kH,
+    Cnn2DKernelSizeLimit l) {
+    if (kW == 1 && kH <= l.vVecLimit) return;
+    if (kH == 1 && kW <= l.hVecLimit) return;
+    if (kH != kW || kH > l.squareLimit || kW > l.squareLimit) {
+        THROW_GNA_EXCEPTION << "Unsupported kerenel in Convolution2D (WxH) " << kW << "x" << kH <<
+            ", only vertical vector up to 1x" << l.vVecLimit << ", horizontal up to " << l.hVecLimit <<
+            "x1 or square up to " << l.squareLimit << "x" << l.squareLimit << " are supported";
+    }
+}
+
+void CheckCnn2DSupported(unsigned IFVC, unsigned kW, unsigned kH, unsigned kN,
+    OvGnaType inPrecision) {
+
+    if (kN % 8 != 0 || kN > 256 || kN == 0) {
+        THROW_GNA_EXCEPTION << "Unsupported number of kernels in Convolution2D: " << kN << ", must be multiply of 8 and in range <8, 256>";
```

Comment:
Typo here and below: should be "multiple".

Reply:
fixed